From 5eefe5baaeb9787ec62eb70dc10e1030ea418415 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 12:58:37 -0400
Subject: [ethernet] qlge: Allow enable/disable rx/tx vlan acceleration independently

Message-id: <1408625927-3701-7-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88534
O-Subject: [PATCH RHEL 7.1 06/16] qlge: Allow enable/disable rx/tx vlan acceleration independently
Bugzilla: 1089357
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>

Bug 1089357 (https://bugzilla.redhat.com/show_bug.cgi?id=1089357)

net commit id 4be1028e9f99011e34dfe405d572592610776c74

o Fix the driver to allow user to enable/disable rx/tx vlan acceleration independently.

  For example:
 ethtool -K ethX rxvlan on/off
 ethtool -K ethX txvlan on/off

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_main.c b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
index 42dac01..e07cbcd 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_main.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
@@ -2376,14 +2376,6 @@ static netdev_features_t qlge_fix_features(struct net_device *ndev,
  netdev_features_t features)
 {
  int err;
- /*
-  * Since there is no support for separate rx/tx vlan accel
-  * enable/disable make sure tx flag is always in same state as rx.
-  */
- if (features & NETIF_F_HW_VLAN_CTAG_RX)
-  features |= NETIF_F_HW_VLAN_CTAG_TX;
- else
-  features &= ~NETIF_F_HW_VLAN_CTAG_TX;
 
  /* Update the behavior of vlan accel in the adapter */
  err = qlge_update_hw_vlan_features(ndev, features);
-- 
1.7.1